### PR TITLE
Misc doc fixes and Unity Object lifetime check bypass fixes

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Fixed documentation errors.
Fixed [Unity.Object lifetime check bypass errors](https://github.com/JetBrains/resharper-unity/wiki/Possible-unintended-bypass-of-lifetime-check-of-underlying-Unity-engine-object)

### CODE REVIEW

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs [hunk 1]
```diff
@@ -126,8 +126,7 @@ public override void Destroy()
         /// <summary>
         /// The <see cref="GameObject"/> to which observed objects are parented.
         /// </summary>
-        private GameObject ObservedObjectParent => observedObjectParent ?? (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent("WindowsMixedRealitySpatialMeshObserver"));
-
+        private GameObject ObservedObjectParent => observedObjectParent != null ? observedObjectParent : (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent("WindowsMixedRealitySpatialMeshObserver"));
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs]: why this change? isn't the original version the pattern established for MRTK?

File: Assets/MixedRealityToolkit/Providers/BaseSpatialAwarenessObserver.cs [hunk 2]
```diff
@@ -96,7 +95,7 @@ public override int GetHashCode()
 
         private GameObject observedObjectParent = null;
 
-        public GameObject ObservedObjectParent => observedObjectParent ?? (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent(Name));
+        public GameObject ObservedObjectParent => observedObjectParent != null ? ObservedObjectParent : (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent(Name));
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Providers/BaseSpatialAwarenessObserver.cs]: are you suggesting this as the new pattern globally within MRTK?

Comment (keveleigh) [reply Assets/MixedRealityToolkit/Providers/BaseSpatialAwarenessObserver.cs]: The difference here is that most of those patterns are referring to interfaces, while this one is a `GameObject`

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs (hunk 1 above)
Comment (keveleigh) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs]: The difference here is that most of those patterns are referring to interfaces, while this one is a `GameObject`, and Unity uses a custom equality check for null which doesn't work in this case.

File: Assets/MixedRealityToolkit/Providers/BaseSpatialAwarenessObserver.cs (hunk 2 above)
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Providers/BaseSpatialAwarenessObserver.cs]: This looks like it'll overflow the stack by calling `ObservedObjectParent` repeatedly.

```suggestion
        public GameObject ObservedObjectParent => observedObjectParent != null ? observedObjectParent : (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent(Name));
```
